### PR TITLE
Disable zkapp commands implementation

### PR DESCRIPTION
**Pull request by ejMina226**
Implementation of https://github.com/MinaProtocol/mina/pull/14880

### CODE REVIEW

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by ejMina226** [comment]
!ci-build-me

File: src/lib/mina_base/user_command.ml [hunk 1]
```diff
@@ -404,24 +404,31 @@ end
...
-  in
-  let errs =
-    match valid_size ~genesis_constants t with
-    | Ok () ->
-        errs0
-    | Error err ->
-        Zkapp_too_big err :: errs0
-  in
-  if List.is_empty errs then Ok () else Error errs
+  if Mina_compile_config.zkapps_disabled then
```
Comment (mrmr1993) [inline src/lib/mina_base/user_command.ml:407]: This logic is not correct, we still need to check for the fee. Add this check as another predicate in `preds` instead.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

File: src/lib/mina_base/user_command.ml (hunk 1 above)
Comment (ejMina226) [reply src/lib/mina_base/user_command.ml]: Thanks for the quick feedback @mrmr1993.

**Comment by ejMina226** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

File: src/lib/mina_base/user_command.ml [hunk 2]
```diff
@@ -391,6 +398,7 @@ module Well_formedness_error = struct
     | Insufficient_fee
     | Zero_vesting_period
     | Zkapp_too_big of (Error.t[@to_yojson Error_json.error_to_yojson])
+    | Zkapps_are_disabled
```
[the remark this answers is not in the document]
Comment (ejMina226) [reply src/lib/mina_base/user_command.ml:401]: Happy to implement your change. You know the codebase better.

**Comment by ejMina226** [comment]
@georgeee Thanks for the review. Could you approve when you're happy?

File: src/lib/mina_base/user_command.ml [hunk 3]
```diff
@@ -400,6 +408,8 @@ module Well_formedness_error = struct
         "Zero vesting period"
     | Zkapp_too_big err ->
         sprintf "Zkapp too big (%s)" (Error.to_string_hum err)
+    | Transaction_type_disabled ->
+        "Zkapps are disabled"
```
Comment (georgeee) [inline src/lib/mina_base/user_command.ml:412]: Now that we changed the name of an error, we should probably update description too

File: src/lib/mina_base/user_command.ml [hunk 4]
```diff
@@ -235,6 +235,13 @@ let minimum_fee = Currency.Fee.minimum_user_command_fee
 
 let has_insufficient_fee t = Currency.Fee.(fee t < minimum_fee)
 
+let is_disabled t =
```
Comment (georgeee) [inline src/lib/mina_base/user_command.ml:238]:
```suggestion
let is_disabled = function
```

and no `match t with` below

Sorry, one last tiny recommendation

**Review by georgeee — APPROVED** [review]

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by georgeee** [comment]
@ejMina226 may I ask you to merge `berkeley` into this branch?

**Comment by ejMina226** [comment]
@georgeee I've merged berkeley in. Looks like there are some issues with the pipeline.

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-nightly-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Comment by deepthiskumar** [comment]
https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/1198

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**